mono/utils/mono-state.c

```diff
@@ -135,6 +137,13 @@ file_for_summary_stage (const char *directory, MonoSummaryStage stage, gchar *bu
 	g_snprintf (buff, sizeof_buff, "%s%scrash_stage_%d", directory, G_DIR_SEPARATOR_S, stage);
 }
 
+void
+mono_summarize_init (const char *timeline_dir)
+{
+	hang_watchdog_dir = mono_config_get_bin_dir ();
```

Comment:
`hang_watchdog_dir` is assigned but not used anywhere.  I think you want a `merp_hang_watchdog_path = g_build_filename (hang_watchdog_dir, "mono-hang-watchdog");` and then use `merp_hang_watchdog_path` in the call to `execv` in `summarizer_supervisor_start`